tests/scripted/check-namespace

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python3
+
+# This script checks the code for namespace related issues
+# Note: This script is not perfect. It can not parse all LMMS files,
+#       and does not contain a complete C++ parser. If you encounter
+#       difficulties with this tests, it could be a fault of your
+#       changes, but it could also be a fault of this script.
+
+import re
+import subprocess
+import sys
+from pathlib import Path
+from typing import NamedTuple
+
+
+# types
+
+class BlockType:
+    pass
+
+
+IF_MACRO = BlockType()
+HEADER_GUARD = BlockType()
+CODE_BLOCK = BlockType()
+EXTERN = BlockType()
```

Comment:
Why comparing this using object identity, not using `Enum`?

Reply:
I checked out Enum, but it didn't give me any better type checking than doing it this way, plus you'd have to write EnumType.MEMBER. Just didn't see the real win...

Reply:
I guess Enum prints nicer error messages though